p2p/src/channels/p2p_channels_effectful_effects.rs

```diff
@@ -0,0 +1,133 @@
+use openmina_core::bug_condition;
+use redux::ActionMeta;
+
+use crate::webrtc::{Offer, P2pConnectionResponse};
+
+use super::{
+    signaling::{
+        discovery::{P2pChannelsSignalingDiscoveryAction, SignalingDiscoveryChannelMsg},
+        exchange::{P2pChannelsSignalingExchangeAction, SignalingExchangeChannelMsg},
+    },
+    ChannelMsg, MsgId, P2pChannelsEffectfulAction, P2pChannelsService,
+};
+
+impl P2pChannelsEffectfulAction {
+    pub fn effects<Store, S>(self, _meta: &ActionMeta, store: &mut Store)
+    where
+        Store: crate::P2pStore<S>,
+        Store::Service: P2pChannelsService,
+    {
+        match self {
+            P2pChannelsEffectfulAction::InitChannel {
+                peer_id,
+                id,
+                on_success,
+            } => {
+                store.service().channel_open(peer_id, id);
+                store.dispatch_callback(on_success, peer_id);
+            }
+            P2pChannelsEffectfulAction::RequestSend {
+                peer_id,
+                msg_id,
+                msg,
+            } => {
+                store.service().channel_send(peer_id, msg_id, msg);
+            }
+            P2pChannelsEffectfulAction::SignalingDiscoveryAnswerDecrypt {
+                peer_id,
+                pub_key,
+                answer,
+            } => {
+                match store
+                    .service()
+                    .decrypt::<P2pConnectionResponse>(&pub_key, &answer)
+                {
+                    Err(_) => {
+                        store.dispatch(P2pChannelsSignalingDiscoveryAction::AnswerDecrypted {
+                            peer_id,
+                            answer: P2pConnectionResponse::SignalDecryptionFailed,
+                        });
+                    }
+                    Ok(answer) => {
+                        store.dispatch(P2pChannelsSignalingDiscoveryAction::AnswerDecrypted {
+                            peer_id,
+                            answer,
+                        });
+                    }
+                }
+            }
+            P2pChannelsEffectfulAction::SignalingDiscoveryOfferEncryptAndSend {
+                peer_id,
+                pub_key,
+                offer,
+            } => {
+                match store.service().encrypt(&pub_key, offer.as_ref()) {
+                    Err(_) => {
+                        // todo!("Failed to encrypt webrtc offer. Handle it.")
```

Comment:
lets make this a `bug_condition` with `TODO` in the message so that it is at least not invisible if it happens

Reply:
@tizoc I'd prefer just logging because this is the error case. It can happen if peer is (most likely) malicious.

Reply:
@binier btw whats the proper way to handle this situation?